src/utilities/SecondMomentAveraging.cpp

```diff
@@ -0,0 +1,249 @@
+
+using namespace amr_wind;
+
+void SecondMomentAveraging::output_line_average_ascii(
+    std::string filename, int step, amrex::Real time)
+{
+    BL_PROFILE("amr-wind::SecondMomentAveraging::output_line_average_ascii");
+
+    if (step != m_last_updated_index) {
+        operator()();
+    }
+
+    if (!amrex::ParallelDescriptor::IOProcessor()) return;
+
+    std::ofstream outfile;
+    outfile.precision(m_precision);
+
+    if (step == 1) {
+
+        // make new file
+        outfile.open(filename.c_str(), std::ios_base::out);
+        outfile << "#ncell,ncomp" << std::endl;
+
+        outfile << m_plane_average1.ncell_line() << ", " << m_num_moments + 3
+                << std::endl;
+        outfile << "#step,time,z";
+
+        for (int m = 0; m < m_plane_average1.ncomp(); ++m) {
+            for (int n = 0; n < m_plane_average1.ncomp(); ++n) {
+                outfile << ",<" + m_plane_average1.field().base_name() +
+                               std::to_string(m) + "'" +
+                               m_plane_average2.field().base_name() +
+                               std::to_string(n) + "'>";
+            }
+        }
+
+        outfile << std::endl;
+
+    } else {
+        // append file
+        outfile.open(filename.c_str(), std::ios_base::out | std::ios_base::app);
+    }
+
+    const int ncomp1 = m_plane_average1.ncomp();
+    const int ncomp2 = m_plane_average2.ncomp();
+
+    for (int i = 0; i < m_plane_average1.ncell_line(); ++i) {
+        outfile << step << ", " << std::scientific << time << ", "
+                << m_plane_average1.line_centroids()[i];
+        for (int m = 0; m < ncomp1; ++m)
+            for (int n = 0; n < ncomp2; ++n)
+                outfile << ", " << std::scientific
+                        << m_second_moments_line
+                               [m_num_moments * i + ncomp1 * m + n];
+
+        outfile << std::endl;
+    }
+}
+
+void SecondMomentAveraging::output_line_average_ascii(
+    int step, amrex::Real time)
+{
+    std::string filename = "second_moment_" + m_plane_average1.field().name() +
+                           "_" + m_plane_average2.field().name() + ".txt";
+    output_line_average_ascii(filename, step, time);
+}
+
+SecondMomentAveraging::SecondMomentAveraging(
+    FieldPlaneAveraging& pa1, FieldPlaneAveraging& pa2)
+    : m_plane_average1(pa1), m_plane_average2(pa2)
+{
+
+    AMREX_ALWAYS_ASSERT(m_plane_average1.axis() == m_plane_average2.axis());
+    AMREX_ALWAYS_ASSERT(m_plane_average1.level() == m_plane_average2.level());
+    AMREX_ALWAYS_ASSERT(
+        m_plane_average1.ncell_plane() == m_plane_average2.ncell_plane());
+    AMREX_ALWAYS_ASSERT(
+        m_plane_average1.ncell_line() == m_plane_average2.ncell_line());
+
+    // this could be relaxed if we wanted one plane to be at a different time
+    // step than another plane
+    AMREX_ALWAYS_ASSERT(
+        m_plane_average1.last_updated_index() ==
+        m_plane_average2.last_updated_index());
+
```

Comment:
this shouldn't be in the constructor. The user could hold references to this object and call `operator()(...)` repeatedly